### 1. Conditional Basics

a. prompt the user for a day of the week, print out whether the day is Monday or not

In [1]:
day_of_the_week = input('Day of the Week: ')
#day_of_the_week = input('Day of the Week: ').lower()

if day_of_the_week == 'Monday' or day_of_the_week == 'monday':
#if day_of_the_week.lower() == 'monday'

    print("It's another wonderful Monday!")
else:
    print(f'{day_of_the_week} is just another day of the week.')

Day of the Week: Doomsday
Doomsday is just another day of the week.


b. prompt the user for a day of the week, print out whether the day is a weekday or a weekend

In [2]:
weekday = ['Monday', 'Mon', 'Tuesday', 'Tues', 'Wednesday', 'Wed', 'Thursday', 'Thur', 'Friday', 'Fri']
weekend = ['Saterday', 'Sat', 'Sunday', 'Sun']
day_of_the_week = input('Day of the Week: ')
if day_of_the_week in weekday:
    print(f'{day_of_the_week} is a weekday.')
elif day_of_the_week in weekend:
    print(f'{day_of_the_week} is on the weekend.')
else:
    print(f'{day_of_the_week} is not a valid day of the week!')

Day of the Week: Doomsday
Doomsday is not a valid day of the week!


c. create variables and make up values for

* the number of hours worked in one week
* the hourly rate
* how much the week's paycheck will be

write the python code that calculates the weekly paycheck. 
You get paid time and a half if you work more than 40 hours.

In [102]:
weekly_hours_worked = 60
hourly_wage = 10
weekly_paycheck = weekly_hours_worked * hourly_wage
overtime = hourly_wage * 1.5

if weekly_hours_worked <= 40: 
    print(weekly_paycheck)

elif weekly_hours_worked > 40: 
    print(weekly_hours_worked * overtime)
    #NEED TO SUBTRACT INITIAL 40hours

900.0


### 2. Loop Basics

a. While

* Create an integer variable i with a value of 5.
* Create a while loop that runs so long as i is less than or equal to 15
* Each loop iteration, output the current value of i, then increment i by one.

Your output should look like this:

5
6
7
8
9
10
11
12
13
14
15

In [4]:
i = int(5)
while i <= 15:
    print(i) # order of print matters
    i += 1

5
6
7
8
9
10
11
12
13
14
15


* Create a while loop that will count by 2's starting with 0 and ending at 100. Follow each number with a new line.

In [7]:
i = 0

while i <= 100:
    print(i)
    i += 2

0
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


* Alter your loop to count backwards by 5's from 100 to -10.

In [1]:
i = 100

while i >= -10 and i <= 100: # order is important!
    # less than 100 listed first will NOT stop at -10
    print(i)
    i -= 5

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5
0
-5
-10


* Create a while loop that starts at 2, and displays the number squared on each line while the number is less than 1,000,000. 

Output should equal:

 2
 4
 16
 256
 65536

In [103]:
number = 2
while number < 1_000_000:
    print(number)
    number **= 2

2
4
16
256
65536


Write a loop that uses print to create the output shown below.

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5

In [4]:
i = 100

while i >= 5 and i <= 100:
    print(i)
    i -= 5

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5


b. For Loops

i. Write some code that prompts the user for a number, then shows a multiplication table up through 10 for that number.

For example, if the user enters 7, your program should output:

7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70

In [25]:
multipliers = map(int, list(range(1,11)))
number = int(input('Choose a whole number: '))
for multiplier in multipliers:
    result = number * multiplier
    print(f'{number} x {multiplier} = {result}')

Choose a whole number: 7
7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70


In [106]:
usernum = input('Choose a whole number: ')
for x in range(1, 11):
    print(f'{usernum} x {x} = {int(usernum * x)}')

Choose a whole number: 7
7 x 1 = 7
7 x 2 = 77
7 x 3 = 777
7 x 4 = 7777
7 x 5 = 77777
7 x 6 = 777777
7 x 7 = 7777777
7 x 8 = 77777777
7 x 9 = 777777777
7 x 10 = 7777777777


NOTE: 1st attempt grabbed input as str

In [31]:
a = input('hg')
for i in range(1,11):
    print(f'{a} * {i} = {a*i}')

hg7
7 * 1 = 7
7 * 2 = 77
7 * 3 = 777
7 * 4 = 7777
7 * 5 = 77777
7 * 6 = 777777
7 * 7 = 7777777
7 * 8 = 77777777
7 * 9 = 777777777
7 * 10 = 7777777777


ii. Create a for loop that uses print to create the output shown below.


1
22
333
4444
55555
666666
7777777
88888888
999999999

In [42]:
a = 1 # a not needed i * i would work
for i in range(1,10):
    print(str(a)*i)
    a += 1

1
22
333
4444
55555
666666
7777777
88888888
999999999


c. break and continue

i. Write a program that prompts the user for a positive integer. Next write a loop that prints out the numbers from the number the user entered down to 1.

In [59]:
pos_int = int(input('Choose a Posititive Intiger: '))
while pos_int > 0:
    print(pos_int)
    pos_int -= 1
    if pos_int == 0:
        break
else:
    print(f'{pos_int} is not a positive integer!')    

Chose a Posititive Intiger: 5
5
4
3
2
1


In [109]:
while True:
    user_num = input('Positive Number: ')
    if user_num.isdigit() == True:
        print('This is a digit')
        if int(user_num) > 0:
            print('Number is Positive')
            break
            
user_num = int(user_num)
for i in range(user_num, 0, -1):
    print(i)

Positive Number: Hello
Positive Number: -7
Positive Number: 5
This is a digit
Number is Positive
5
4
3
2
1


ii. The input function can be used to prompt for input and use that input in your python code. Prompt the user to enter a positive number and write a loop that counts from 0 to that number. 

(Hints: first make sure that the value the user entered is a valid number, also note that the input function returns a string, so you'll need to convert this to a numeric type.)

In [111]:
pos_int = int(input('Choose a Posititive Intiger: '))
number = 0
while pos_int > 0:
    print(number)
    number += 1
    if number > pos_int:
        break
else:
    print(f'{pos_int} is not a positive integer!')    

Choose a Posititive Intiger: 5
0
1
2
3
4
5


In [110]:
while True:
    user_num = input('Positive Number: ')
    if user_num.isdigit() == True:
        print('This is a digit')
        if int(user_num) > 0:
            print('Number is Positive')
            break
            
for i in range(int(user_num) +1):
    print(i)

Positive Number: 5
This is a digit
Number is Positive
0
1
2
3
4
5


iii. Prompt the user for an odd number between 1 and 50. 
Use a loop and a break statement to continue prompting the user if they enter invalid input. 

(Hint: use the isdigit method on strings to determine this). 

Use a loop and the continue statement to output all the odd numbers between 1 and 50, except for the number the user entered.

Your output should look like this:

Number to skip is: 27

Here is an odd number: 1
Here is an odd number: 3
Here is an odd number: 5
Here is an odd number: 7
Here is an odd number: 9
Here is an odd number: 11
Here is an odd number: 13
Here is an odd number: 15
Here is an odd number: 17
Here is an odd number: 19
Here is an odd number: 21
Here is an odd number: 23
Here is an odd number: 25
Yikes! Skipping number: 27
Here is an odd number: 29
Here is an odd number: 31
Here is an odd number: 33
Here is an odd number: 35
Here is an odd number: 37
Here is an odd number: 39
Here is an odd number: 41
Here is an odd number: 43
Here is an odd number: 45
Here is an odd number: 47
Here is an odd number: 49

In [86]:
user_input = input('Choose an odd number between 1 and 50: ')
if odd_num = user_input.isdigit() % 2 == 1
odd_1_to_50 = odd_num > 0 and odd_num <= 50
print(odd_1_to_50)

#while odd_num :
#    if odd_1_to_50 % 2 == 0:
#     print(odd_1_to_50)
#    else:
#        print(f'{odd_1_to_50} is not an an odd number between 1 and 50!')
#        number += 2
#    if odd_1_to_50 > 50:
#        break
#else:
#    print(f'{odd_1_to_50} is not an an odd number between 1 and 50!')

Choose an odd number between 1 and 50: 3
True


In [125]:
while True:
    user_num = input('Enter an odd number between 1 and 50: ')
    
    if user_num.isdigit():
        #print('This is a digit')
        if int(user_num) % 2 != 0:
            #print('This is an odd number')
            if (int(user_num) > 1) and (int(user_num) < 50):
                #print('This number is between 1-50')
                break
                
user_num = int(user_num)            
                
for i in range(1, 50):
    if i == user_num:
        print(f'Yikes! Skipping number: {user_num}')
        continue
    if i % 2 == 1:
        print(f'Here is an odd number: {i}')

Enter an odd number between 1 and 50: 20
Enter an odd number between 1 and 50: 27
Here is an odd number: 1
Here is an odd number: 3
Here is an odd number: 5
Here is an odd number: 7
Here is an odd number: 9
Here is an odd number: 11
Here is an odd number: 13
Here is an odd number: 15
Here is an odd number: 17
Here is an odd number: 19
Here is an odd number: 21
Here is an odd number: 23
Here is an odd number: 25
Yikes! Skipping number: 27
Here is an odd number: 29
Here is an odd number: 31
Here is an odd number: 33
Here is an odd number: 35
Here is an odd number: 37
Here is an odd number: 39
Here is an odd number: 41
Here is an odd number: 43
Here is an odd number: 45
Here is an odd number: 47
Here is an odd number: 49


3. Fizzbuzz

One of the most common interview questions for entry-level programmers is the FizzBuzz test. Developed by Imran Ghory, the test is designed to test basic looping and conditional logic skills.

* Write a program that prints the numbers from 1 to 100.
* For multiples of three print "Fizz" instead of the number
* For the multiples of five print "Buzz".
* For numbers which are multiples of both three and five print "FizzBuzz".

* Write a program that prints the numbers from 1 to 100.

In [93]:
imposter = 1
while imposter > 0 and imposter <= 100:
    print(imposter)
    imposter += 1


1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100


In [128]:
for i in range(1,101):
    if i % 15 == 0:
        print(f'{i} - FizzBuzz')
        continue
    if i % 3 == 0:
        print (f'{i} - Fizz')
        continue
    if i % 5 == 0:
        print(f'{i} - Buzz')
        continue
    print(i)

1
2
3 - Fizz
4
5 - Buzz
6 - Fizz
7
8
9 - Fizz
10 - Buzz
11
12 - Fizz
13
14
15 - FizzBuzz
16
17
18 - Fizz
19
20 - Buzz
21 - Fizz
22
23
24 - Fizz
25 - Buzz
26
27 - Fizz
28
29
30 - FizzBuzz
31
32
33 - Fizz
34
35 - Buzz
36 - Fizz
37
38
39 - Fizz
40 - Buzz
41
42 - Fizz
43
44
45 - FizzBuzz
46
47
48 - Fizz
49
50 - Buzz
51 - Fizz
52
53
54 - Fizz
55 - Buzz
56
57 - Fizz
58
59
60 - FizzBuzz
61
62
63 - Fizz
64
65 - Buzz
66 - Fizz
67
68
69 - Fizz
70 - Buzz
71
72 - Fizz
73
74
75 - FizzBuzz
76
77
78 - Fizz
79
80 - Buzz
81 - Fizz
82
83
84 - Fizz
85 - Buzz
86
87 - Fizz
88
89
90 - FizzBuzz
91
92
93 - Fizz
94
95 - Buzz
96 - Fizz
97
98
99 - Fizz
100 - Buzz


* For multiples of three print "Fizz" instead of the number

In [94]:
imposter = 1
while imposter > 0 and imposter <= 100:
    if imposter % 3 == 0:
        print('Fizz')
    print(imposter)
    imposter += 1

1
2
Fizz
3
4
5
Fizz
6
7
8
Fizz
9
10
11
Fizz
12
13
14
Fizz
15
16
17
Fizz
18
19
20
Fizz
21
22
23
Fizz
24
25
26
Fizz
27
28
29
Fizz
30
31
32
Fizz
33
34
35
Fizz
36
37
38
Fizz
39
40
41
Fizz
42
43
44
Fizz
45
46
47
Fizz
48
49
50
Fizz
51
52
53
Fizz
54
55
56
Fizz
57
58
59
Fizz
60
61
62
Fizz
63
64
65
Fizz
66
67
68
Fizz
69
70
71
Fizz
72
73
74
Fizz
75
76
77
Fizz
78
79
80
Fizz
81
82
83
Fizz
84
85
86
Fizz
87
88
89
Fizz
90
91
92
Fizz
93
94
95
Fizz
96
97
98
Fizz
99
100


4. Display a table of powers.

* Prompt the user to enter an integer.
* Display a table of squares and cubes from 1 to the value entered.
* Ask if the user wants to continue.
* Assume that the user will enter valid data.
* Only continue if the user agrees to.

Example Output
What number would you like to go up to? 5

Here is your table!

number | squared | cubed
------ | ------- | -----
1      | 1       | 1
2      | 4       | 8
3      | 9       | 27
4      | 16      | 64
5      | 25      | 125

Bonus: Research python's format string specifiers to align the table

In [129]:
user_num = int(input('Choose an intiger: '))
user_num, user_num**2, user_num**3

Choose an intiger: 3


(3, 9, 27)

In [152]:
while True:
    user_num = int(input('Choose an integer: '))

    for i in range(1, user_num + 1):
        print(f'{i} |{i**2} |{i**3}')
    
    user_yn = input('Would you like to continue? y/n: ')
    if user_yn.lower() != 'y':
        break

Choose an integer: 5
1 |1 |1
2 |4 |8
3 |9 |27
4 |16 |64
5 |25 |125
Would you like to continue? y/n: y
Choose an integer: 8
1 |1 |1
2 |4 |8
3 |9 |27
4 |16 |64
5 |25 |125
6 |36 |216
7 |49 |343
8 |64 |512
Would you like to continue? y/n: n


5. Convert given number grades into letter grades.

* Prompt the user for a numerical grade from 0 to 100.
* Display the corresponding letter grade.
* Prompt the user to continue.
* Assume that the user will enter valid integers for the grades.
* The application should only continue if the user agrees to.
* Grade Ranges:

    * A : 100 - 88
    * B : 87 - 80
    * C : 79 - 67
    * D : 66 - 60
    * F : 59 - 0

Bonus

Edit your grade ranges to include pluses and minuses (ex: 99-100 = A+).

In [140]:
while True:
    user_grade = int(input('Enter a numerical grade 1-100'))
    if user_grade >=88:
        print('A')
    elif user_grade >=80:
        print('B')
    elif user_grade >=67:
        print('C')
    elif user_grade >=60:
        print('D')
    else:
        print('F')
    
    user_yn = input('Would you like to continue? y/n: ')
    if user_yn.lower() != 'y':
        break

Enter a numerical grade 1-10056
F
Would you like to continue? y/n: y
Enter a numerical grade 1-10089
A
Would you like to continue? y/n: n


LEARN DICTIONARY FORMAT AND FUNCTION!

6. Create a list of dictionaries where each dictionary represents a book that you have read. Each dictionary in the list should have the keys title, author, and genre. Loop through the list and print out information about each book.

a. Prompt the user to enter a genre, then loop through your books list and print out the titles of all the books in that genre.

In [161]:
list_of_dict = {'title': 'Title-01', 'author': 'Author-01', 'genre': 'Genre-01',
                'title': 'Title-02', 'author': 'Author-02', 'genre': 'Genre-02',
                'title': 'Title-03', 'author': 'Author-03', 'genre': 'Genre-03'
               }
for title in list_of_dict:
    print(title)

title
author
genre
